File: leetcode-solutions/0605-can-place-flowers/solution.py
class Solution(object):
    def canPlaceFlowers(self, flowerbed, n):
        """
        :type flowerbed: List[int]
        :type n: int
        :rtype: bool
        """
        for i in range(len(flowerbed)):
            if flowerbed[i]==0:
                left=(i==0 or flowerbed[i-1]==0)
                right=(i==(len(flowerbed)-1) or flowerbed[i+1]==0)
                if left and right:
                    flowerbed[i]=1
                    n-=1
        return n<=0
        

File: leetcode-solutions/0605-can-place-flowers/solution.java
class Solution {
    public boolean canPlaceFlowers(int[] flowerbed, int n) {
        int l=flowerbed.length;
        for(int i=0;i<l;i++){
            if(flowerbed[i]==0){
                boolean left=(i==0 || flowerbed[i-1]==0);
            boolean right=(i==(l-1) || flowerbed[i+1]==0);
            if(left && right){
                flowerbed[i]=1;
                n-=1;
            }
            }
        }
        return n<=0;
    }
}
